server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -109,32 +109,28 @@ public Query toQuery(Function function, Context context) {
             if (!ref.isNullable()) {
                 return new MatchNoDocsQuery("`x IS NULL` on column that is NOT NULL can't match");
             }
-            Query refExistsQuery = refExistsQuery(ref, context, true);
+            Query refExistsQuery = refExistsQuery(ref, context);
             return refExistsQuery == null ? null : Queries.not(refExistsQuery);
         }
         return null;
     }
 
 
     @Nullable
-    public static Query refExistsQuery(Reference ref, Context context, boolean countEmptyArrays) {
+    public static Query refExistsQuery(Reference ref, Context context) {
```

Comment:
`countEmptyArrays` can be removed because when `ref` is an array type, `countEmptyArrays` passed in is always `true`.

Reply:
Do I get it right, that we needed these parameter because it used to be called in array_length and in https://github.com/crate/crate/pull/16329 we changed `ArrayUpperFunction` to not use it anymore?

Reply:
Yes, that were 3 call sites that called `refExistsQuery` with `countEmptyArrays` as `false`, `ArrayUpperFunction` and `AnyNeqOperator`. The first two were removed by #16329 and the other is irrelevant since `ref` is never an array type as the method name suggests, `AnyNeqOperator.refMatchesAnyArrayLiteral`(`ref != ANY(<array_literal>)`).